### PR TITLE
OCW course sync fixes

### PR DESCRIPTION
**Pull request by alizag**
#### What's this PR do?
Updates requirements
Changes ocw sync to ignore all courses under QA/.
Fixes issue with colliding course prefixes

### CODE REVIEW

File: course_catalog/tasks.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def get_ocw_data():
             course_instance = None
         try:
             # fetch JSON contents for each course file in memory (slow)
-            for obj in raw_data_bucket.objects.filter(Prefix=course_prefix):
+            for obj in raw_data_bucket.objects.filter(Prefix=course_prefix + "/"):
```
Comment (mbertrand) [inline course_catalog/tasks.py]: You might potentially end up filtering by a `Prefix` with two trailing slashes here, if `course_prefix` is assigned to `key_pieces[0] + "/"` a few lines up.

Maybe change to `(Prefix=course_prefix)`, and include the trailing slash in the `course_prefix` value if necessary?

File: course_catalog/tasks.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def get_ocw_data():
     log.info("Assembling list of courses...")
     for file in raw_data_bucket.objects.all():
         key_pieces = file.key.split("/")
-        course_prefix = key_pieces[0] + "/" + key_pieces[1]
+        course_prefix = key_pieces[0] + "/" + key_pieces[1] if key_pieces[0] == "PROD" else key_pieces[0] + "/"
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/tasks.py]: Added

File: course_catalog/constants.py [hunk 3]
```diff
@@ -26,8 +26,7 @@ class PlatformType(Enum):
     "PROD/mathematics",
     "PROD/more",
     "PROD/physics",
-    "QA/do-not-publish",
-    "QA/engineering"
+    "QA/"
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/constants.py]: Done

File: course_catalog/tasks.py (hunk 1 above)
Comment (sesterki) [reply course_catalog/tasks.py]: Done

**Pull request merged**